knowledge-base/accumulate-expression-from-group-total-in-overall-total.md

```diff
@@ -0,0 +1,119 @@
+---
+title: Accumulate an Expression Value from Group Total in an Overall Total
+description: How to calculate the overall total of an expression that represents group total
+type: how-to
+page_title: Form an Overall Total from a Group Total Expression
+slug: accumulate-expression-from-group-total-in-overall-total
+position: 
+tags: 
+ticketid: 1528624
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Progress® Telerik® Reporting</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+Let's describe the problem with a simple scenario. Assume that you have to create a report for workers where you need to calculate the total hours per pay week for each. 
+If this total is greater than 40 hrs/week, then its value becomes 40 hrs and the rest of the hours are passed as overtime hours. 
+Both the total and the overtime hours for the particular group/scope can be easily calculated with an [Expression](../report-expressions).
+The problem comes if you need to sum the values of the overtime total hours in the Summary of the report. That said, to create a total of the tolal overtime hours for 
+all the groups.  
+
+The most straight-forward idea would be to sum up the values in the _TextBox_ displaying the total for the group, i.e. from all instances of this _TextBox_. 
+Unfortunately, by design, the value in a particular report item  cannot be accessed from another item. On the other hand, the overtime hours are not present 
+in the data source and are rather calculated in an _Expression_, hence utilizing [Global Objects](../expressions-global-objects) directly is not an option. 
+
+## Solution
+You need to create a [Custom Aggregate Function](../expressions-user-aggregate-functions) that accumulates the overtime hours from the groups. 
+Here is a sample implementation for the case when workers reports are grouped by weeks:
+
+```C#
+using System;
+using System.Collections.Generic;
+using Telerik.Reporting.Expressions;
+
+namespace CustomAggregates
+{
+    public class TotalOvertimeHours : IAggregateFunction
+    {
+        private Dictionary<long, TimeSpan> WorkTimeByWeek;
```

Comment:
if we need to show proper code style to the clients, I would go with **WorkTimeByWeek -> workTimeByWeek** and also use **this.** prefix in each usage.